Change characteristic "area" to "radius"

SeanCurtis-TRI opened the pull request:
Bad dimensional analysis snuck through PRs. This updates the *discussion*
of the model parameter. It changes the incorrect "area" to be a characteristic "radius".

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/7568)
<!-- Reviewable:end -->

SeanCurtis-TRI commented:
+@sherm1 for feature review please -- the most important focus would be on the `contact_model_doxygen.h`. Everything else is largely superficial -- renaming area to radius.

---

Review status: 0 of 20 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-:-L-dVUrmnHsxOcgExIhY:b-4f91qw)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
-@sherm1 +@edrumwri for feature review instead, please.

---

Review status: 0 of 20 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-:-L-dW-qOSKbpEyk0PwAF:b-18ozk5)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
/cc @siyuanfeng-tri because you caught it.

---

Review status: 0 of 20 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-:-L-dWdQedludv_ivqwSo:boh9vgb)*
<!-- Sent from Reviewable.io -->

siyuanfeng-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 20 of 20 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 44 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-goT3yBIHaTYlVY_XC:-L-goT3yBIHaTYlVY_XD:b-y0ub45) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/manipulation/util/world_sim_tree_builder.cc#L44)):*
> ```cpp
>   //     large.
>   contact_model_parameters_.v_stiction_tolerance = 0.01;  // m/s
>   contact_model_parameters_.characteristic_radius = 1.0;  // m^2
> ```

btw.. why is this number huge cmp to the others??

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-:-L-gqM4nyKnVKoAyoFpy:bnfp4nl)*
<!-- Sent from Reviewable.io -->

edrumwri commented:
Reviewed 20 of 20 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 236 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-gei67o2XBMqwDrJSQ:-L-gei69mc_VVaZfUQ2C:b-3c5vuj) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L236)):*
> ```objc
>     produce larger contact forces. For example, a 1 cm rubber ball on a table
>     has a much smaller contact patch than the tires of a car would have on the
>     road. Even if we assume the ball and tires are made of the same rubber, the
> ```

This example is flawed, because of course the "forces in play are quite different": the contact forces acting on a car tire (presumably attached to a car) are very different than the forces acting on a much lighter tennis ball. Why not use two balls as an example with the same mass, but one large and one small?

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 607 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-gfpjA82AxzHfK4v5Z:-L-gfpjC5ElSfQbGLJ8c:b-btqvtq) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L607)):*
> ```objc
>  tunable "global" parameter of the
>  @ref drake::systems::CompliantContactModel "CompliantContactModel".
>  (See @ref contact_engineering on how to work with this property.)
> ```

I know a group of folks discussed this at length before it went in originally, but I have one significant concern with this text. If I'm a user attempting to set parameters for my bodies, what is the implication of this cylinder-on-plane approximation for the shapes in my simulation?  As an example, how far off will this prediction be for two spheres? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/manipulation/util/world_sim_tree_builder.cc, line 44 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-goT3yBIHaTYlVY_XC:-L-ldS3Lm9cPosRGLXD4:b-khuqrg) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/manipulation/util/world_sim_tree_builder.cc#L44)):*
<details><summary><i>Previously, siyuanfeng-tri wrote…</i></summary><blockquote>

btw.. why is this number huge cmp to the others??
</blockquote></details>

OK Because it reflected the original values. The goal in adding the per-element contact material is to make it so that every simulation was the same after the change as before. So, if the simulation had changed to some "non-standard" values, the mapped values are likewise non-standard (albeit transformed).

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 236 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-gei67o2XBMqwDrJSQ:-L-ldhT6l8t20N5bpxzz:b7f2asq) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L236)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

This example is flawed, because of course the "forces in play are quite different": the contact forces acting on a car tire (presumably attached to a car) are very different than the forces acting on a much lighter tennis ball. Why not use two balls as an example with the same mass, but one large and one small?
</blockquote></details>

Two things:
  - I'd hoped that in context, we were only suggesting contact forces in this discussion and, even then, normal forces at that.
  - We've got specific examples with cars that need to be cranked up, so I figured my making this analogy, it would be clear why, at least, one specific example was tuned the way it was.

The balls are abstract and I can put in enough legalese to attempt to make an incontrovertibly correct statement -- but I don't know if it would make this documentation any more helpful.

Let me know how strongly you feel about this.

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 607 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-gfpjA82AxzHfK4v5Z:-L-leNtKpd_ygl01IxYS:b-czgglj) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L607)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

I know a group of folks discussed this at length before it went in originally, but I have one significant concern with this text. If I'm a user attempting to set parameters for my bodies, what is the implication of this cylinder-on-plane approximation for the shapes in my simulation?  As an example, how far off will this prediction be for two spheres? 
</blockquote></details>

Those are good questions, but the purpose of this documentation is *not* to answer those questions. That's why any of the technical details are left to the bottom. The best you can get is a vague intuition of working with the current compliant contact model -- turn this value up for bigger, stiffer forces. Set this value to something you look up in a table. This third value is a sloppy tuning parameter, etc. I wouldn't advocate bounding the error of the implementation because then we'd be obliged to document the error of the model as well.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
+@sherm1 for platform review
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 20 of 20 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[drake/examples/contact_model/bowling_ball.cc, line 85 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-ng97aHsi29f2tTP75:-L-ng97aHsi29f2tTP76:b24jcgw) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/examples/contact_model/bowling_ball.cc#L85)):*
> ```cpp
>   cout << "\tdynamic friction: " << FLAGS_ud << "\n";
>   cout << "\tslip threshold:   " << FLAGS_v_tol << "\n";
>   cout << "\tContact area:     " << FLAGS_contact_radius << "\n";
> ```

BTW text still says "area"

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 359 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-nhs-HOl9KnzNQQJNz:-L-nhs-HOl9KnzNQQJO-:b-24fmii) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L359)):*
> ```objc
>  approaching speed), and the geometry and material parameters of the contacting
>  bodies. Notice there is no approximation in this equation; it is always true
>  since this equation _is_ the definition of the average contact pressure `p(q)`.
> ```

BTW p(q,v)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-:-L-npNA4gmuQXQpQ0d4I:b-yccdo8)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
This is waiting for you @edrumwri 

---

Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-:-L-npbsupmNcy381E-GS:b-eaibkv)*
<!-- Sent from Reviewable.io -->

edrumwri commented:
I know- I'm reading the whole thing fresh to see if it feels better to me.

---

Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 236 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-gei67o2XBMqwDrJSQ:-L-o3hhsIjmrVCJ376EX:bb6a3hp) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L236)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

Two things:
  - I'd hoped that in context, we were only suggesting contact forces in this discussion and, even then, normal forces at that.
  - We've got specific examples with cars that need to be cranked up, so I figured my making this analogy, it would be clear why, at least, one specific example was tuned the way it was.

The balls are abstract and I can put in enough legalese to attempt to make an incontrovertibly correct statement -- but I don't know if it would make this documentation any more helpful.

Let me know how strongly you feel about this.
</blockquote></details>

I don't understand- the contact forces are exactly the issue. The contact forces- normal forces- will be dependent upon contact patch, mass, mass distribution, velocities, and more. I thought the example was trying to explain that, all other factors being equal, the contact forces would be bigger for a larger contact patch.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-:-L-oBGGBiJudJ_ILuwUj:b-24iwzf)*
<!-- Sent from Reviewable.io -->

edrumwri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> (more like Looks Mildly Acceptable to Me)

---

Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 38 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-oBnwI-LD6-rRUMegs:-L-oBnwI-LD6-rRUMegt:bl6mjy6) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L38)):*
> ```objc
> 
>  This document discusses a _compliant_ contact model. In compliant models, the
>  bodies are considered deformable. The
> ```

BTW I read the statement "the bodies are considered deformable" and my hackles went up. As you note below, this particular compliant model is applied to a rigid body, and the compliant model is only supposed to be a very thin layer around a rigid core. My immediate thought upon reading that "the bodies are considered deformable" is that this effects the dynamics as well. 

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 383 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-s9sMrgYvcBgZpL1DT:-L-s9sMs0SI1a6XIj9yy:bezktkm) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L383)):*
> ```objc
>  Consider two spheres made of the same material and of the same size (with
>  Young's modulus E and radius R). If the amount that the two spheres are
>  penetrating are x, then the resultant normal force (according to the Hertz
> ```

BTW Nit: "are" -> "is"

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 607 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-gfpjA82AxzHfK4v5Z:-L-o4zovVYlAtiM5je1n:bulsnr3) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L607)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

Those are good questions, but the purpose of this documentation is *not* to answer those questions. That's why any of the technical details are left to the bottom. The best you can get is a vague intuition of working with the current compliant contact model -- turn this value up for bigger, stiffer forces. Set this value to something you look up in a table. This third value is a sloppy tuning parameter, etc. I wouldn't advocate bounding the error of the implementation because then we'd be obliged to document the error of the model as well.
</blockquote></details>

Ugh. You've written this up as well as could be done, I think, but it's unsatisfying.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-:-L-sAe1LF8YBbXID1Gks:bi0bfv2)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
Rebased on master and addressed editorial comments. @edrumwri you still have an outstanding comment.

---

Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[drake/examples/contact_model/bowling_ball.cc, line 85 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-ng97aHsi29f2tTP75:-L-sEB5WxAwXgTiywCmh:b-lot626) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/examples/contact_model/bowling_ball.cc#L85)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW text still says "area"
</blockquote></details>

Done (and in `sliding_bricks.cc`

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 38 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-oBnwI-LD6-rRUMegs:-L-sEJVtf6d2jX46SYWS:bb6un1j) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L38)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

BTW I read the statement "the bodies are considered deformable" and my hackles went up. As you note below, this particular compliant model is applied to a rigid body, and the compliant model is only supposed to be a very thin layer around a rigid core. My immediate thought upon reading that "the bodies are considered deformable" is that this effects the dynamics as well. 
</blockquote></details>

Done

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 236 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-gei67o2XBMqwDrJSQ:-L-sF6mW7IqBjwSdarN-:b-mnnu1v) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L236)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

I don't understand- the contact forces are exactly the issue. The contact forces- normal forces- will be dependent upon contact patch, mass, mass distribution, velocities, and more. I thought the example was trying to explain that, all other factors being equal, the contact forces would be bigger for a larger contact patch.
</blockquote></details>

I've re-worded it, let me know what you think.

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 359 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-nhs-HOl9KnzNQQJNz:-L-sFGRjKUKCi22BUotF:bb6un1j) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L359)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW p(q,v)
</blockquote></details>

Done

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 383 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-s9sMrgYvcBgZpL1DT:-L-sFLN46lBnV8kKavQY:bb6un1j) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L383)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

BTW Nit: "are" -> "is"
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-:-L-sFPWZRha8tIVAs9pc:b4kamv1)*
<!-- Sent from Reviewable.io -->

edrumwri commented:
Reviewed 1 of 40 files at r2.
Review status: 1 of 20 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/multibody/rigid_body_plant/contact_model_doxygen.h, line 236 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7568#-L-gei67o2XBMqwDrJSQ:-L-scT8VpvbtsrAawOhB:b-85i63v) ([raw file](https://github.com/robotlocomotion/drake/blob/960b0b7001eabbf5fa59fa2781423fad8dc23c44/drake/multibody/rigid_body_plant/contact_model_doxygen.h#L236)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

I've re-worded it, let me know what you think.
</blockquote></details>

LGTM.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7568)*
<!-- Sent from Reviewable.io -->
